frontend/frontend/protocols/ssh/_transport_manager.py

```diff
@@ -0,0 +1,140 @@
+import datetime
+import logging
+import threading
+from time import sleep
+from typing import List, NamedTuple
+
+import paramiko
+from frontend.config import config
+from ._proxy_handler import ProxyHandler
+from ._ssh_server import Server
+
+
+logger = logging.getLogger(__name__)
+
+
+class TransportPair(
+    NamedTuple("TransportPair",
+               [("attacker_transport", paramiko.Transport),
+                ("proxy_handler", ProxyHandler),
+                ("server", Server)])):
+    """Class for holding the attacker transport and the ProxyHandler as a tuple"""
+
+
+class TransportManager:
```

Comment:
This class keeps track of all open SSH connections and makes sure to call the relevant end methods if it detects that:
- The attacker connection is not active anymore
- The attacker connection has no open channel and has been inactive for x minutes